filecoin-proofs/src/api/post.rs

```diff
@@ -88,13 +89,26 @@ impl PrivateReplicaInfo {
     }
 
     /// Generate the merkle tree of this particular replica.
-    pub fn merkle_tree(&self, sector_size: u64) -> Result<Tree, Error> {
-        let mut f_in = File::open(&self.access)?;
-        let mut data = Vec::new();
-        f_in.read_to_end(&mut data)?;
-
-        let bytes = PaddedBytesAmount(sector_size as u64);
-        public_params(bytes, 1).graph.merkle_tree(&data)
+    pub fn merkle_tree(&self, _sector_size: u64) -> Result<Tree, Error> {
+        let tree_size = {
```

Comment:
We can just compute the size of the tree based on the `sector_size` argument, no need to open the file.